data/adapters/gersbach_E2G_CRISPR_adapter.py

```diff
@@ -0,0 +1,119 @@
+        if label not in GersbachE2GCRISPR.ALLOWED_LABELS:
+            raise ValueError('Ivalid label. Allowed values: ' +
+                             ','.join(GersbachE2GCRISPR.ALLOWED_LABELS))
+        self.data_file = filepath
+        self.guide_file = reference_filepath
+        self.label = label
+        self.source_url = source_url
+        self.file_accession = source_url.split('/')[-2]
+        self.dataset = label
+        self.type = 'edge'
+        if (self.label == 'genomic_element'):
+            self.type = 'node'
+        self.writer = writer
+
+    def process_file(self):
+        self.writer.open()
+        guide_rna_sequences = parse_guide_file(self.guide_file)
+        genomic_elements = {}
+        guide_id_to_element_id = {}
+        for guide_id, guide_rna in guide_rna_sequences.items():
+            name = guide_rna.get('intended_target_name')
+            gene = name.split('.')[0] if name.startswith('ENSG') else None
+            chr = guide_rna.get('intended_target_chr')
+            start = guide_rna.get('intended_target_start')
+            end = guide_rna.get('intended_target_end')
+            if gene and chr and start is not None and end is not None:
+                element_id = build_regulatory_region_id(
+                    chr, start, end, 'CRISPR'
+                )
+                guide_id_to_element_id[guide_id] = element_id
+                if element_id not in genomic_elements:
+                    genomic_elements[element_id] = {
+                        'gene': gene, 'chr': chr, 'start': start, 'end': end}
+        if self.label == 'genomic_element':
+            for genomic_element in genomic_elements:
+                element_type = 'promoter'
+                if genomic_elements[genomic_element]['end'] - genomic_elements[genomic_element]['start']:
+                    element_type = 'transcription start site'
+                _id = genomic_element + '_' + self.file_accession
+                _props = {
+                    '_key': _id,
+                    'name': _id,
+                    'chr': genomic_elements[genomic_element]['chr'],
+                    'start': genomic_elements[genomic_element]['start'],
+                    'end': genomic_elements[genomic_element]['end'],
+                    'method_type': 'CRISPR',
+                    'type': element_type,
+                    'source': GersbachE2GCRISPR.SOURCE,
+                    'source_url': self.source_url,
+                    'files_filesets': 'files_filesets/' + self.file_accession
+                }
+                self.writer.write(json.dumps(_props))
+                self.writer.write('\n')
+        elif self.label == 'genomic_element_gene':
+            ffs = FileFileSet(accessions=[], writer=None,
+                              label='igvf_file_fileset')
+            file_set_props, _, _ = ffs.query_fileset_files_props_igvf(
+                self.file_accession, replace=False)
```

Comment:
I'm going to remove replace in required input arg: https://github.com/IGVF-DACC/igvf-catalog/pull/437/files#diff-7673406be69c3fdd484e608a206e1da97b02017b36063adb57bb9f77cc087d36 if that makes sense to you

Reply:
yeah you're right it's not necessary to pass it in